README.md

```diff
@@ -1,2 +1,307 @@
+ ---> 8b7eb0124777
+Step 8/13 : ADD public/index.html /app/public/index.html
+ ---> 9d763716676c
+Step 9/13 : ADD public/script.js /app/public/script.js
+ ---> 353bca474993
+Step 10/13 : ADD public/style.css /app/public/style.css
+ ---> 81819bed4684
+Step 11/13 : WORKDIR /app
+ ---> 1a99524a116a
+Removing intermediate container 020c16388591
+Step 12/13 : CMD ./guestbook
+ ---> Running in bdb88f981da7
+ ---> fc76ec26b945
+Removing intermediate container bdb88f981da7
+Step 13/13 : EXPOSE 3000
+ ---> Running in b04f65bdedad
+ ---> b63df6f0716e
+Removing intermediate container b04f65bdedad
+Successfully built b63df6f0716e
+Successfully tagged registry.ng.bluemix.net/guestbook/gbv1:2018-05-18-1
+The push refers to a repository [registry.ng.bluemix.net/guestbook/gbv1]
+cfd8baa1be3c: Pushed
+3d8c58449645: Pushed
+98a83cd4e753: Pushed
+9b5fbbfcbb1d: Pushed
+5f70bf18a086: Layer already exists
+5dbcf0efe4f2: Layer already exists
+4: digest: sha256:cca8cdb32ed616ae13087a9b557696896505bb36241c0f24588987094a687000 size: 1772
+
+OK
+```
+
+This builds and pushes all the relevant layers.
+
+The `--tag` field gives the image a name and version, separated by the
+`:`. When using the IBM Cloud Container Registry the name is going to
+include the full URL to that image.
+
+The version portion can be anything that's meaningful to you. The
+important thing is that tags do not get reused. If you rebuild an
+image, make sure to change the version before uploading. Many
+kubernetes primitives, such as deployments, will perform a graceful
+rolling upgrade when you specify a new image version. If image
+versions get reused it hampers the effectiveness of container
+orchestration systems.
+
+# Retrieving from IBM Cloud
+
+IBM Cloud images are included in kubernetes configuration in the same
+way as images from docker hub. The only difference is the full url is
+needed as it's the non default image repository.
+
+```
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: guestbook
+  labels:
+    app: guestbook
+spec:
+  replicas: 3
+  selector:
+    matchLabels:
+      app: guestbook
+  template:
+    metadata:
+      labels:
+        app: guestbook
+    spec:
+      containers:
+      - name: guestbook
+        image: registry.ng.bluemix.net/guestbook/gbv1:2018-05-18-1
+        ports:
+        - name: http-server
+          containerPort: 3000
+```
+
+# Vulnerability Advisor
+
+An additional feature of using the IBM Container Service is IBM
+Vulnerabiliy Advisor. After the image is uploaded it is scanned by a
+set of tools that look for common security issues. These scans are
+also run regularly as new CVEs are issued.
+
+In the case of the guestbook image it's an extremely small
+microservice that doesn't include a traditional OS base, so there is
+not much VA can do with it. If we look at a set of images that are
+based on an Ubuntu Linux base we can see it in action.
+
+```
+> bx cr images --restrict ny-power
```

Comment:
It doesn't show anything useful

Reply:
```
> bx cr va registry.ng.bluemix.net/guestbook/gbv1:2018-05-22-1 -e
Checking vulnerabilities for 'registry.ng.bluemix.net/guestbook/gbv1:2018-05-22-1'...

Image 'registry.ng.bluemix.net/guestbook/gbv1:2018-05-22-1' was last scanned on Tue May 22 15:04:01 UTC 2018.
No vulnerability assessment is available because the image operating system is not supported by Vulnerability Advisor.

CONFIGURATION ISSUES FOUND
==========================

SECURITY PRACTICE                           CORRECTIVE ACTION   
File /etc/login.defs not found              Maximum password age must be set to 90 days.   
File /etc/pam.d/common-password not found   Minimum password length must be 8.   
File /etc/login.defs not found              Minimum days that must elapse between user-initiated password changes should be 1.   

```